src/soc-support.md

```diff
@@ -0,0 +1,235 @@
+Thorough testing ensures hardware-software reliability, and leveraging these
+tools can significantly enhance development workflows.
+
+## Nice-to-have features for enhanced ecosystem support
+
+### Examples
+
+Including some basic examples as part of the HAL is essential for helping
+developers get started. These examples should demonstrate key functionalities,
+such as initializing peripherals or handling interrupts. They serve as
+practical starting points and learning aids.
+
+### BSP (Board Support Package) crates
+
+BSP crates are relevant when you need to provide board-specific configurations
+and initializations. Unlike HALs, which focus on hardware abstraction, BSPs
+handle the integration of multiple components for a specific board. Separation
+in BSP and HAL crates offers a layered approach, making it easier for developers
+to build applications targeting a particular hardware board.
+
+### Project templates
+
+Project templates are boilerplate code structures that provide a starting point
+for new projects. They include prevalent configurations, dependencies, and
+setup steps, saving developers time and reducing the learning curve. Examples
+of project templates include bare-metal (using the HAL without any framework),
+Embassy, RTIC, and others.
+
+### Integration with popular IDEs and tools
+
+Offer guides on setting up development environments for Embedded Rust projects
+with popular tools such as:
+
+- [rust-analyzer](https://rust-analyzer.github.io): for Rust syntax
+  highlighting and error checking.
+- [probe-rs](https://probe.rs): for flashing and debugging firmware.
+- [defmt](https://crates.io/crates/defmt): a logging framework optimized for
+  embedded systems, including a test harness called
+  [defmt-test](https://crates.io/crates/defmt-test).
+
+Providing setup instructions for these tools will help developers integrate
+them into their workflows, enhancing productivity and collaboration.
+
+## Suggested flow for adding SoC Support
+
+- A preliminary requirement of this flow is that the Rust toolchain includes
+  a [target](https://doc.rust-lang.org/rustc/platform-support.html) that
+  matches the System-on-Chip (SoC). If this not the case the solution can be as
+  simple as adding a
+  [custom target](https://doc.rust-lang.org/rustc/targets/custom.html) or as
+  difficult as adding support for the underlying architecture to
+  [LLVM](https://llvm.org).
+- Before starting from scratch, check if any existing community efforts for
+  already exist (e.g. checking on
+  [awesome-embedded-rust](https://github.com/rust-embedded/awesome-embedded-rust)
+  or joining the
+  [Rust Embedded Matrix room](https://matrix.to/#/#rust-embedded:matrix.org)).
+  This could save significant development time.
+- Ensure that your target is supported by [probe-rs](https://probe.rs). The
+  ability to debug using SWD or JTAG is highly beneficial. Support for flashing
+  programming can be added with a Flash Algorithm (e.g. from a CMSIS-Pack).
```

Comment:
It might be worth pointing to the probe-rs flash algorithm template here as well: https://github.com/probe-rs/flash-algorithm-template 

If a CMSIS pack algo doesn't exist, this is a pretty easy way to implement flash algorithms in rust.

Reply:
That's a great suggestion! @mbrossard want to go ahead and add this before we merge?

Reply:
Done